noodles-fasta/src/reader.rs

```diff
@@ -306,6 +310,57 @@ where
     Ok(bytes_read)
 }
 
+fn read_segment<R>(reader: &mut R, buf: &mut Vec<u8>, max_bases: usize) -> io::Result<usize>
+where
+    R: BufRead,
+{
+    const LINE_FEED: u8 = b'\n';
+    const CARRIAGE_RETURN: u8 = b'\r';
+
+    let mut bases_left = max_bases;
+    let mut bytes_read = 0;
+
+    loop {
+        if bases_left == 0 {
+            break;
+        }
+        let reader_buf = reader.fill_buf()?;
+
+        if reader_buf.is_empty() || reader_buf[0] == DEFINITION_PREFIX {
+            break;
+        }
+
+        let len = match memchr(LINE_FEED, reader_buf) {
+            Some(i) => {
+                let i = std::cmp::min(i, bases_left);
+                bases_left -= i;
```

Comment:
This doesn't compensate for `CRLF` newlines. E.g., `AC\r\nGT` would return `i = 3` when there are 2 bases.

Reply:
Thanks, good catch. Please consider my change, also reproduced below for convenience.

```diff
diff --git a/noodles-fasta/src/reader.rs b/noodles-fasta/src/reader.rs
index 12acdfb2..10e2b067 100644
--- a/noodles-fasta/src/reader.rs
+++ b/noodles-fasta/src/reader.rs
@@ -333,14 +333,15 @@ where
         let len = match memchr(LINE_FEED, reader_buf) {
             Some(i) => {
                 let i = std::cmp::min(i, bases_left);
-                bases_left -= i;
                 let line = &reader_buf[..i];

                 if line.ends_with(&[CARRIAGE_RETURN]) {
                     let end = line.len() - 1;
                     buf.extend(&line[..end]);
+                    bases_left -= end;
                 } else {
                     buf.extend(line);
+                    bases_left -= i;
                 }

                 i + 1
```